assets/js/modules/idea-hub/components/dashboard/DashboardIdeasWidget/Idea.js

```diff
@@ -40,40 +41,62 @@ import {
 	IDEA_HUB_BUTTON_UNPIN,
 	IDEA_HUB_BUTTON_DELETE,
 	IDEA_HUB_BUTTON_VIEW,
+	IDEA_HUB_ACTIVITY_CREATING_DRAFT,
+	IDEA_HUB_ACTIVITY_DRAFT_CREATED,
+	IDEA_HUB_ACTIVITY_IS_PROCESSING,
 } from '../../../datastore/constants';
 import DeleteIcon from '../../../../../../svg/idea-hub-delete.svg';
 import CreateIcon from '../../../../../../svg/idea-hub-create.svg';
 import PinIcon from '../../../../../../svg/idea-hub-pin.svg';
 import UnpinIcon from '../../../../../../svg/idea-hub-unpin.svg';
 
-const { useDispatch } = Data;
+const DRAFT_CREATED_TIMER = 2000;
 
-const Idea = ( { postEditURL, name, text, topics, buttons } ) => {
+const { useDispatch, useSelect } = Data;
+
+const Idea = ( props ) => {
+	const { postEditURL, name, text, topics, buttons } = props;
 	const {
 		createIdeaDraftPost,
 		saveIdea,
 		unsaveIdea,
 		dismissIdea,
+		setActivity,
+		removeActivity,
+		removeIdeaFromNewAndSavedIdeas,
 	} = useDispatch( STORE_NAME );
-	const [ isProcessing, setIsProcessing ] = useState( false );
+	const activity = useSelect( ( select ) => select( STORE_NAME ).getActivity( name ) );
 
 	const handleDelete = useCallback( async () => {
+		setActivity( name, IDEA_HUB_ACTIVITY_IS_PROCESSING );
 		await dismissIdea( name );
-	}, [ name, dismissIdea ] );
+		removeActivity( name );
+	}, [ name, dismissIdea, setActivity, removeActivity ] );
 
 	const handlePin = useCallback( async () => {
+		setActivity( name, IDEA_HUB_ACTIVITY_IS_PROCESSING );
 		await saveIdea( name );
-	}, [ name, saveIdea ] );
+		removeActivity( name );
+	}, [ name, saveIdea, setActivity, removeActivity ] );
 
 	const handleUnpin = useCallback( async () => {
+		setActivity( name, IDEA_HUB_ACTIVITY_IS_PROCESSING );
 		await unsaveIdea( name );
-	}, [ name, unsaveIdea ] );
+		removeActivity( name );
+	}, [ name, unsaveIdea, setActivity, removeActivity ] );
 
 	const handleCreate = useCallback( async () => {
-		setIsProcessing( true );
+		setActivity( name, IDEA_HUB_ACTIVITY_CREATING_DRAFT );
+
 		await createIdeaDraftPost( { name, text, topics } );
-		setIsProcessing( false );
-	}, [ name, text, topics, createIdeaDraftPost ] );
+
+		setActivity( name, IDEA_HUB_ACTIVITY_DRAFT_CREATED );
+
+		setTimeout( () => {
+			removeActivity( name );
+			removeIdeaFromNewAndSavedIdeas( props );
```

Comment:
@danielgent let's pass the idea name to the `removeIdeaFromNewAndSavedIdeas` action since it is the only thing that we need there.

Reply:
This is for tree-shaking to be done better right?

I copied this syntax for elsewhere so it could be worth doing this across the whole codebase :smile: